### PR TITLE
Galactic code review

### PR DESCRIPTION
**Pull request by cesar-lopez-mar**
Made most changes suggested in PR25 https://github.com/nobleo/full_coverage_path_planner/pull/25

### CODE REVIEW

File: src/full_coverage_path_planner.cpp [hunk 1]
```diff
@@ -69,9 +62,8 @@ namespace full_coverage_path_planner
   {
     geometry_msgs::msg::PoseStamped new_goal;
     std::list<Point_t>::const_iterator it, it_next, it_prev;
-    int dx_now, dy_now, dx_next, dy_next, move_dir_now = 0, move_dir_next = 0;
     bool do_publish = false;
-    float orientation = eDirNone;
+    float orientation = dir::none;
     RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Received goalpoints with length: %lu", goalpoints.size());
     if (goalpoints.size() > 1)
```
Comment (cesar-lopez-mar) [inline src/full_coverage_path_planner.cpp:68]: Can you elaborate more on your previous [comment](https://github.com/nobleo/full_coverage_path_planner/pull/25#discussion_r677243381)?

Comment (Timple) [reply src/full_coverage_path_planner.cpp:68]: Sure!

Early-outs make the code more readable by handling the exceptions in a short block at the top.

```
function do_awesome()
{
  if (everything_okay_as_expected)
  {
    Do 
    lots 
    of 
    indented stuff
  }
  else
  {
    warn about it
  }
}
```
Can be replaced by:

```
function do_awesome()
{
  if (!everything_okay_as_expected)
  { 
    warn about it
    return;
  }

  Do 
  lots 
  of 
  stuff
}
```

Comment (Timple) [reply src/full_coverage_path_planner.cpp:68]: And here it looks like `goalpoints.size() <= 1` is exceptional.

Comment (cesar-lopez-mar) [reply src/full_coverage_path_planner.cpp:68]: Got it... in our case the function does not return right after the else condition, so we can not early-out. I can switch the order to at least show the especial case first.

Comment (Timple) [reply src/full_coverage_path_planner.cpp:68]: Aaah, I thought it did!
Then you may leave it like this as well. Up to you.

Comment (Timple) [reply src/full_coverage_path_planner.cpp:68]: On a similar note. I think the node dies if `goalpoints.size() == 0`. Since that case isn't handled.

Comment (cesar-lopez-mar) [reply src/full_coverage_path_planner.cpp:68]: Yep.. will add that one

**Comment by cesar-lopez-mar** [comment]
ping

File: include/full_coverage_path_planner/full_coverage_path_planner.hpp [hunk 2]
```diff
@@ -38,17 +38,19 @@ using std::string;
...
-  eDirRight = 1,
-  eDirUp = 2,
-  eDirLeft = -1,
-  eDirDown = -2,
-};
 
 namespace full_coverage_path_planner
 {
+
+  enum dir
```
Comment (Timple) [inline include/full_coverage_path_planner/full_coverage_path_planner.hpp:45]: `enum class dir`

**Comment by Timple** [comment]
These improvements all look good, so approving.

But this PR is of course only a review of the improvements, not a full review. So that issue stays open.

I'll pick that up later this week.

**Review by Timple — APPROVED** [review]

**Pull request merged**